src/toolchain.rs

```diff
@@ -1,29 +1,58 @@
 use anyhow::{bail, Result};
 use std::fmt;
+use std::fs::{remove_dir_all, remove_file};
 use std::path::PathBuf;
 use std::str::FromStr;
 use tracing::info;
 
+use crate::component;
 use crate::download::{download_file_and_unpack, link_to_fuelup, unpack_bins, DownloadCfg};
 use crate::ops::fuelup_self::self_update;
-use crate::path::{ensure_dir_exists, fuelup_bin, fuelup_bin_dir, toolchain_bin_dir};
+use crate::path::{
+    ensure_dir_exists, fuelup_bin, fuelup_bin_dir, settings_file, toolchain_bin_dir,
+};
+use crate::settings::SettingsFile;
+
+pub const RESERVED_TOOLCHAIN_NAMES: &[&str] = &["latest", "nightly"];
 
 pub enum DistToolchainName {
     Latest,
 }
 
+impl fmt::Display for DistToolchainName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            DistToolchainName::Latest => write!(f, "latest"),
+        }
+    }
+}
+
 impl FromStr for DistToolchainName {
     type Err = anyhow::Error;
     fn from_str(s: &str) -> Result<Self> {
-        let split = s.split_once('-').unwrap();
-        let name = split.0;
+        let name = s.split_once('-').map(|n| n.0);
         match name {
-            "latest" => Ok(Self::Latest),
+            Some("latest") => Ok(Self::Latest),
             _ => bail!("Unknown name for toolchain: {}", s),
         }
     }
 }
 
+pub fn is_official_toolchain(toolchain: &str) -> bool {
+    let mut reserved: Vec<String> = Vec::new();
```

Comment:
It looks like we shouldn't really need to collect a `reserved` `Vec` here as the `Vec` is never returned and is only used to check if a name exists. Instead, perhaps we can just return directly if we come across a reserved string? E.g. something like:

```rust
for reserved_toolchain_name in RESERVED_TOOLCHAIN_NAMES {
    if toolchain == reserved_toolchain_name {
        return true;
    } else if let Some(triple) = triple {
        if toolchain == reserved_toolchain_name + triple {
            return true;
        }
    }
}
```

Btw just checking - should we be adding a `-` between the toolchain name and the triple?

Reply:
Ah yes - we should be, good spot here

Reply:
Actually, it seems like I stopped using this for whatever reason. I actually do not recall but it might've been for the `fuelup toolchain new <toolchain>` command. I'm going to remove this for now